### PR TITLE
Fix for https://jira.mongodb.org/browse/JAVA-2648

### PR DESCRIPTION
**Pull request by bcluap**
Allow a POJO without a setter for a collection to have the collection
populated via the getter and addAll()

### CODE REVIEW

**Comment by bcluap** [comment]
Any chance of this being pulled? Tx

**Comment by rozza** [comment]
Hi @bcluap,

Firstly, thanks for the PR, apologies for the lack of response. We are currently focusing on MongoDB 3.6 features for the upcoming release of the Java driver.  I have some PojoCodec tickets to review in due course and I will review this PR as part of that work.

Ross

File: bson/src/main/org/bson/codecs/pojo/PropertyAccessorImpl.java [hunk 1]
```diff
@@ -55,6 +57,16 @@
                 } else {
                     propertyMetadata.getField().set(instance, value);
                 }
+            } else if (propertyMetadata.getGetter() != null
```
Comment (72MiguelGomes) [inline bson/src/main/org/bson/codecs/pojo/PropertyAccessorImpl.java]: Why not directly call isSerializable(), this way you also have the logic around transient fields...

Comment (bcluap) [reply bson/src/main/org/bson/codecs/pojo/PropertyAccessorImpl.java]: Something like this?:

```java
} else if (propertyMetadata.isSerializable() && value instanceof Collection) {
                // PCB : https://jira.mongodb.org/browse/JAVA-2648
                // If a type is a collection and there is no setter then use the getter and addAll to it from the value
                // Also check that the collection is currently empty. Without this check some test cases fail.
                T collection;
                if (propertyMetadata.getGetter() != null) {
                    collection = (T) propertyMetadata.getGetter().invoke(instance);
                } else {
                    collection = (T) propertyMetadata.getField().get(instance);
                }
                Collection<?> col = ((Collection<?>) collection);
                if (col != null && col.isEmpty()) {
                    col.addAll((Collection) value);
                }
            }
```

**Comment by rozza** [comment]
Closing as support for setting collections / maps via getters has been added via a convention.

**Pull request closed without merging**